packages/pools/src/concentrated/fetch-pool-amount-provider.ts

```diff
@@ -0,0 +1,71 @@
+import { Int } from "@keplr-wallet/unit";
+
+import { AmountsDataProvider, ConcentratedLiquidityPool } from "./pool";
+
+type BankAmountResponse = {
+  balances: {
+    amount: string;
+    denom: string;
+  }[];
+};
+
+/** Provider of bank balances from bank module by bech32Address. Expected by concentrated liquidity pool. */
+export class FetchPoolAmountProvider implements AmountsDataProvider {
+  /** Address => Response */
+  protected amountsCache = new Map<
+    string,
+    Awaited<ReturnType<AmountsDataProvider["getPoolAmounts"]>>
+  >();
+
+  constructor(
+    protected readonly baseNodeUrl: string,
+    protected readonly poolId: string,
+    protected readonly cacheDurationMs = 1000 * 30, // 30 seconds
```

Comment:
Can you define as a constant?

Reply:
It is, as a default param

Reply:
My point is I don't think these in particular are worth extracting elsewhere since they are certainly only used in that one place. I do think constants at the top of file or in a file have their place.

But, when developers rely on their convention/habit of looking at the top of the file, they should only find variables that are important and heavily relied on. Especially those found in dedicated files (which at that point I think are config variables, which generally should be minimized). Otherwise, we risk losing signal to noise.